OrbitLinuxTracing/TracerThread.cpp

```diff
@@ -155,165 +155,59 @@ void TracerThread::Run(
         // PERF_RECORD_SAMPLE, PERF_RECORD_MMAP, etc., defined in enum
         // perf_event_type in perf_event.h.
         switch (header.type) {
-          // non system-wide profiling:
-          case PERF_RECORD_SWITCH: {
-            ContextSwitchPerfEvent event;
-            ring_buffer.ConsumeRecord(header, &event.ring_buffer_record);
-            if (listener_ != nullptr) {
-              if (event.IsSwitchOut()) {
-                ContextSwitchOut context_switch_out{
-                    event.GetTid(), static_cast<uint16_t>(event.GetCpu()),
-                    event.GetTimestamp()};
-                listener_->OnContextSwitchOut(context_switch_out);
-              } else {
-                ContextSwitchIn context_switch_in{
-                    event.GetTid(), static_cast<uint16_t>(event.GetCpu()),
-                    event.GetTimestamp()};
-                listener_->OnContextSwitchIn(context_switch_in);
-              }
-            }
-            ++sched_switch_count;
-          } break;
-
-          // system-wide profiling
-          case PERF_RECORD_SWITCH_CPU_WIDE: {
-            SystemWideContextSwitchPerfEvent event;
-            ring_buffer.ConsumeRecord(header, &event.ring_buffer_record);
-
-            if (listener_ != nullptr) {
-              if (event.GetPrevTid() != 0) {
-                ContextSwitchOut context_switch_out{
-                    event.GetPrevTid(), static_cast<uint16_t>(event.GetCpu()),
-                    event.GetTimestamp()};
-                listener_->OnContextSwitchOut(context_switch_out);
-              }
-              if (event.GetNextTid() != 0) {
-                ContextSwitchIn context_switch_in{
-                    event.GetNextTid(), static_cast<uint16_t>(event.GetCpu()),
-                    event.GetTimestamp()};
-                listener_->OnContextSwitchIn(context_switch_in);
-              }
-            }
-
-            ++sched_switch_count;
-          } break;
-
-          case PERF_RECORD_FORK: {
-            ForkPerfEvent event;
-            ring_buffer.ConsumeRecord(header, &event.ring_buffer_record);
-
-            if (event.GetPid() == pid_) {
-              // A new thread of the sampled process was spawned.
-              int sampling_fd = sample_mmap_task_event_open(sampling_period_ns_,
```

Comment:
The reason fds_to_ring_buffer_to_add was used and the ring buffer was opened here was to start collecting events from the new thread as soon as the event was seen rather than at the end of the loop. I don't know if this makes any difference and it probably doesn't in the vast majority of cases, but still, that was the reason for the complication instead of using new_threads_ how it is done now.

Reply:
Reverted to the original code.